### ISSUE TITLE
ddbug prints the wrong strings for object file.

### ISSUE THREAD

**Issue by jrmuizel** [gimli-rs/ddbug#2]
Running ddbug on https://people-mozilla.org/~jmuizelaar/tmp/webrender-710bba83bc2d83b7.0.o.gz

prints 'rustc version 1.18.0-nightly (2564711e8 2017-04-04)' for every string.

**Comment by jrmuizel** [comment]
It looks like this is a gimli issue as I can reproduce the same problem with gimli's dwarfdump. The problem seems related to relocations.

**Comment by jrmuizel** [comment]
This problem is already known: https://github.com/gimli-rs/gimli/issues/180

**Comment by m4b** [comment]
I'd suggest to perhaps leave this open; just encountered this myself and didn't see an issue, and almost opened a new one.

Strangely, this only seems to affect ELF object files, and the correct output seems more or less present for mach-o object files cross-compiled on a linux system, e.g. something compiled as follows on a host linux system:

```
rustc -g --emit=obj --target=x86_64-apple-darwin test.rs -o test.macho
```

**Comment by philipc** [comment]
Implemented in 2589d8f
